packages/@sfx/sayt-driver-plugin/test/unit/common/sayt-driver-plugin.test.ts

```diff
@@ -370,141 +390,63 @@ describe('Sayt Driver Plugin', () => {
-          allMeta: {
-            title: 'some-title',
-            visualVariants: [{
-              productImage: 'some-link',
-              nonvisualVariants: [{
-                originalPrice: 3.99,
-              }],
-            }],
+      response = {
+        records: [
+          {
+            allMeta: {
+              title: firstProductTitle,
+            },
           },
-        }],
-      };
-      const expectedResponse = {
-        products: [
           {
-            title: 'some-title',
-            price: 3.99,
-            imageSrc: 'some-link',
-            imageAlt: 'some-title',
-            productUrl: 'some-link',
+            allMeta: {
+              title: secondProductTitle,
+            },
           },
         ],
-        originalResponse: response,
       };
-
-      const actualProduct = driver.searchCallback(response);
-
-      expect(actualProduct).to.deep.equal(expectedResponse);
     });
 
-    it('should filter out an incomplete product object', () => {
-      const goodObject = {
-        allMeta: {
-          title: 'some-title',
-          visualVariants: [{
-            productImage: 'some-link',
-            nonvisualVariants: [{
-              originalPrice: 3.99,
-            }],
-          }],
-        },
-      };
+    it('should return a response with products', () => {
+      const result = driver.searchCallback(response);
 
-      const badObject1 = {
-        allMeta: {},
-      };
+      expect(result.products).to.have.lengthOf(2);
+      expect(result.products[0].allMeta.title).to.equal(firstProductTitle);
+      expect(result.products[1].allMeta.title).to.equal(secondProductTitle);
+    });
 
-      const badObject2 = {
-        allMeta: {
-          title: 'some-title',
-          visualVariants: [],
-        },
-      };
+    it('should map products using the transform function', () => {
+      const key2 = 'value2';
+      driver.transformProduct = (product) => ({
+        key1: product.allMeta.title,
+        key2,
+      });
 
-      const expectedGoodObject = {
-        title: 'some-title',
-        price: 3.99,
-        imageSrc: 'some-link',
-        imageAlt: 'some-title',
-        productUrl: 'some-link',
-      };
+      const result = driver.searchCallback(response);
 
-      const input = {
-        records: [
-          goodObject,
-          goodObject,
-          badObject1,
-          badObject2,
-          badObject2,
-          goodObject,
-        ],
-      };
+      expect(result.products).to.deep.equal([
+        { key1: firstProductTitle, key2 },
+        { key1: secondProductTitle, key2 },
+      ]);
+    });
 
-      const expectedResponse = {
-        products: [
-          expectedGoodObject,
-          expectedGoodObject,
-          expectedGoodObject,
-        ],
-        originalResponse: input,
+    it('should filter out any products that map to falsy values', () => {
+      driver.transformProduct = (product, i) => {
+        if (i == 1) return;
```

Comment:
I'm surprised the linter didn't complain about the usage of `==` instead of `===`.

Reply:
That's unlike me! Fixed.